model/infinispan/src/main/java/org/keycloak/models/cache/infinispan/ClientAdapter.java

```diff
@@ -53,7 +53,7 @@ public ClientAdapter(RealmModel cachedRealm, CachedClient cached, RealmCacheSess
     private void getDelegateForUpdate() {
         if (updated == null) {
             cacheSession.registerClientInvalidation(cached.getId(), cached.getClientId(), cachedRealm.getId());
-            updated = cacheSession.getRealmDelegate().getClientById(cached.getId(), cachedRealm);
+            updated = cacheSession.getClientDelegate().getClientById(cached.getId(), cachedRealm);
```

Comment:
```suggestion
            updated = cacheSession.getClientDelegate().getClientById(cachedRealm, cached.getId());
```
We should use new non-deprecated method instead. 

Newly depricated `ClientLookupProvider.getClientById(String, RealmModel)`, `ClientLookupProvider.getClientByClientId(String, RealmModel)` and `ClientLookupProvider.searchClientsByClientId(String, Integer, Integer, RealmModel)` are used in several places in codebase including testsuite.